core/src/main/java/com/scalar/db/storage/dynamo/PutStatementHandler.java

```diff
@@ -58,46 +57,40 @@ private void execute(Put put, TableMetadata tableMetadata) {
     DynamoMutation dynamoMutation = new DynamoMutation(put, tableMetadata);
     String expression;
     String condition = null;
-    Map<String, String> expressionAttributeNameMap = new HashMap<>();
+    Map<String, String> expressionAttributeNameMap;
     Map<String, AttributeValue> bindMap;
 
     if (!put.getCondition().isPresent()) {
       expression = dynamoMutation.getUpdateExpressionWithKey();
-      expressionAttributeNameMap.putAll(dynamoMutation.getColumnMapWithKey());
+      expressionAttributeNameMap = dynamoMutation.getColumnMapWithKey();
       bindMap = dynamoMutation.getValueBindMapWithKey();
     } else if (put.getCondition().get() instanceof PutIfNotExists) {
       expression = dynamoMutation.getUpdateExpressionWithKey();
-      expressionAttributeNameMap.putAll(dynamoMutation.getColumnMapWithKey());
+      expressionAttributeNameMap = dynamoMutation.getColumnMapWithKey();
       bindMap = dynamoMutation.getValueBindMapWithKey();
       condition = dynamoMutation.getIfNotExistsCondition();
     } else if (put.getCondition().get() instanceof PutIfExists) {
       expression = dynamoMutation.getUpdateExpression();
       condition = dynamoMutation.getIfExistsCondition();
-      expressionAttributeNameMap.putAll(dynamoMutation.getColumnMap());
+      expressionAttributeNameMap = dynamoMutation.getColumnMap();
       bindMap = dynamoMutation.getValueBindMap();
     } else {
       expression = dynamoMutation.getUpdateExpression();
-      condition = dynamoMutation.getCondition();
-      expressionAttributeNameMap.putAll(dynamoMutation.getColumnMap());
+      condition = dynamoMutation.getIfExistsCondition() + " AND " + dynamoMutation.getCondition();
+      expressionAttributeNameMap = dynamoMutation.getColumnMap();
+      expressionAttributeNameMap.putAll(dynamoMutation.getConditionColumnMap());
```

Comment:
Should we do combining maps into a new map instead of mutating a map from an existing one getting from another place?

Reply:
Thank you for asking that. Yes, that's a good point. Actually, it's basically not a good practice to mutate a map returned from another class's method because we don't know the implementation of the map (ex., it might be an immutable one). I mutate the map here because we already do the same thing in some other parts (ex., https://github.com/scalar-labs/scalardb/blob/dbe49adc20ab88f4d53fe68ab1c057557238abbe/core/src/main/java/com/scalar/db/storage/dynamo/PutStatementHandler.java#L82-L83), and we know `DynamoMutation.getColumnMap()` returns a mutable `HashMap` since `DynamoMutation` is our class. 

What do you think? If no objection, I'll merge it. Thanks.

Reply:
I see. Let's merge it. Thanks!